src/Http/Services/ResourcesGetter.php

```diff
@@ -39,23 +39,22 @@ public function perform() {
         $query->where(function($query) { $this->addFilters($query); });
 
         $this->records = $query->get();
+    }
 
-        $queryCount = $this->model->select($this->tableNameModel.'.*');
-        $this->addJoins($queryCount);
-        $queryCount->where(function($query) { $this->addSearch($query); });
-        $queryCount->where(function($query) { $this->addFilters($query); });
+    public function count() {
+        $query = $this->getBaseQuery();
 
-        $this->recordsCount = $queryCount->count();
+        $this->recordsCount = $query->count();
     }
 
-    public function getQueryForBatch() {
-      $query = $this->model->select($this->tableNameModel.'.*');
+    public function getBaseQuery() {
```

Comment:
This function should not be `public`, use `protected` instead.
Can't you use this function in the `perform` to prevent the code duplication?